File: js-pr/56/javascript/code-quality-performance/string_counter_45.js
// {fact rule=code-quality-performance@v1.0 defects=0}
function countCharacters(text) {
    return [...text].reduce((counts, char) => {
        counts[char] = (counts[char] || 0) + 1;
        return counts;
    }, {});
}
// {/fact}

File: java-pr/49/java/performance/StringCounter45.java
// {fact rule=performance@v1.0 defects=0}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class StringCounter45 {
    private Database database = new Database();
    
    public Map<Character, Integer> countCharacters(String text) {
        return text.chars()
                .mapToObj(c -> (char) c)
                .collect(Collectors.groupingBy(c -> c, Collectors.summingInt(c -> 1)));
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}